src/coreclr/dlls/mscorpe/pewriter.cpp

```diff
@@ -614,6 +614,12 @@ HRESULT PEWriter::Init(PESectionMan *pFrom, DWORD createFlags)
         m_ntHeaders->FileHeader.Characteristics |= VAL16(IMAGE_FILE_RELOCS_STRIPPED);
     }
 
+    if (createFlags & ICEE_CREATE_FILE_DET)
+    {
+        // A timestamp of 0 triggers asserts in the VM
+        m_ntHeaders->FileHeader.TimeDateStamp = VAL32(1);
```

Comment:
https://github.com/dotnet/runtime/blob/main/docs/design/specs/PE-COFF.md and the behavior of `csc -deterministic` use a hash of the file in this field.  I see other hashing going on in this file, but I think it applies to other parts of the file (?)

It's not clear to me if this matters.  I'm not sure what the difference is between this field and the value can be stored in the REPRO debug directory.

I wondered if the runtime is asserting because it wants (mostly?) uniqueness of these values, but if https://github.com/dotnet/runtime/blob/f7334fab4babc2cef606fafc053b4e5abe4edf80/src/coreclr/vm/dwbucketmanager.hpp#L651 is the assertion that you're referring to, then it only seems to be about the value zero, which is unfortunate since https://learn.microsoft.com/en-us/windows/win32/debug/pe-format seems to indicate that 0 and FFFFFFFF are valid.  (Perhaps FFFFFFFF would be a better choice if producing a hash is problematic for some reason?)

Reply:
That is the assert I was hitting. The [deterministic PE/COFF documentation](https://github.com/dotnet/runtime/blob/main/docs/design/specs/PE-COFF.md#deterministic-pecoff-file) suggests the actual timestamp value doesn't matter, though I suppose we ought to match `csc`'s behavior and use the file hash here too, rather than some magic value we have to explain. Either way, I agree that the assertion in the VM is probably overzealous.